Source/Engine/Content/Factories/IAssetFactory.h

```diff
@@ -67,3 +67,32 @@ class FLAXENGINE_API IAssetFactory
         return nullptr;
     }
 };
+
+#if NDEBUG
+#define INTERNAL_FACTORY_REGISTER_CHEAK(type)\
+if (IAssetFactory::Get().ContainsKey(type::TypeName)) \
+{ \
+    if (Platform::IsDebuggerPresent()) \
+    { \
+    PLATFORM_DEBUG_BREAK; \
+    } \
+    Platform::Assert("[IAssetFactory] Can't register "#type \
+        " it is already registered", __FILE__, __LINE__); \
+        return; \
+}
+#define INTERNAL_FACTORY_UNREGISTER_CHEAK(type)\
+if (!IAssetFactory::Get().ContainsKey(type::TypeName)) \
```

Comment:
CHECK(IAssetFactory::Get().ContainsKey(type::TypeName))

Reply:
custom massage better suits this case 
giving better information was is wrong